File: types/invoice.ts
import type { MinorAmount } from "@/types/money";

export type InvoiceStatus =
  | "DRAFT"
  | "ISSUED"
  | "AWAITING_PAYMENT"
  | "PAYMENT_DETECTED"
  | "PAID_ON_CHAIN"
  | "SETTLEMENT_PENDING"
  | "PAYOUT_PROCESSING"
  | "PAID_OUT"
  | "EXPIRED"
  | "UNDERPAID"
  | "OVERPAID"
  | "WRONG_TOKEN"
  | "MANUAL_REVIEW"
  | "CANCELLED";

export interface InvoiceDraft {
  contractorId: string;
  description: string;
  sourceAmountMinor: MinorAmount;
  sourceCurrency: "USDC";
  dueDate: string;
}

export interface Invoice extends InvoiceDraft {
  id: string;
  organizationId: string;
  invoiceNumber: string;
  paymentRequestId: string;
  status: InvoiceStatus;
  createdAt: string;
}

File: backend/src/main/java/com/nova/backend/domain/Invoice.java
package com.nova.backend.domain;

import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;

public record Invoice(
    UUID id,
    UUID organizationId,
    String contractorId,
    String invoiceNumber,
    String description,
    String amountMinor,
    String currency,
    LocalDate dueDate,
    String status,
    Instant createdAt
) {}
